manifests/patch-e2e.sh

```diff
@@ -38,20 +32,32 @@ namespace=${namespace:-tidb-admin}
 CURDIR=$(cd $(dirname ${BASH_SOURCE[0]}); pwd )
 os=`uname -s| tr '[A-Z]' '[a-z]'`
 
-CA_BUNDLE=$(kubectl get configmap -n kube-system extension-apiserver-authentication -o=jsonpath='{.data.client-ca-file}' | base64 | tr -d '\n')
-echo $CA_BUNDLE
+case ${os} in
+  linux)
+    sed -i "s/namespace: \${NAMESPACE}/namespace: ${namespace}/g" $CURDIR/webhook.yaml
+    ;;
+  darwin)
+    sed -i "" "s/namespace: \${NAMESPACE}/namespace: ${namespace}/g" $CURDIR/webhook.yaml
+    ;;
+    *)
+    echo "invalid os ${os}, only support Linux and Darwin" >&2
+    ;;
+esac
 
+if [ "$bundle" = true ] ; then
+
+CA_BUNDLE=$(kubectl get configmap -n kube-system extension-apiserver-authentication -o=jsonpath='{.data.client-ca-file}' | base64 | tr -d '\n')
 
 case ${os} in
   linux)
     sed -i "s/caBundle: .*$/caBundle: ${CA_BUNDLE}/g" $CURDIR/webhook.yaml
-    sed -i "s/namespace: \${NAMESPACE}/namespace: ${namespace}/g" $CURDIR/webhook.yaml
```

Comment:
why move this line to 37?

Reply:
There are 2 `sed` actions in this script. The first `sed` would change the namespace for `webhook.yaml`. And the second `sed` would change the `cabundle: null` into `cabundle: <cabundle token>` depends on kube-apiserver version.